src/Future.ts

```diff
@@ -0,0 +1,62 @@
+import * as Async from "./Async";
+import * as Fx from "./Fx";
+
+export interface Future<E, A> {
+  readonly state: FutureState<E, A>;
+  readonly wait: Fx.Fx<Async.Async | E, A>;
```

Comment:
We could potentially start with Future and Queue being internal, and possibly implement them directly using promises.  That's what I did for Semaphore.  They can always be exposed later.  Any thoughts on that?

Reply:
I don't have any issues with that. I really only hacked them into place to get withEmitter working in some capacity